codegen/smithy-aws-ruby-codegen/src/main/java/software/amazon/smithy/aws/ruby/codegen/AWSConfig.java

```diff
@@ -58,6 +61,102 @@ public final class AWSConfig {
+            """;
+    public static final ClientConfig SESSION_TOKEN = ClientConfig.builder()
+            .name("session_token")
+            .documentationRbsAndValidationType("String")
+            .documentation(SESSION_TOKEN_DOCUMENTATION)
+            .defaults(ConfigProviderChain.builder()
+                    .dynamicProvider("('stubbed-token' if cfg[:stub_responses])")
+                    .build())
+            .build();
+
+    private static final String CREDENTIALS_PROVIDER_DOCUMENTATION = """
+            A credentials provider is a class that fetches your AWS credentials and responds to the `#identity` 
+            method. This can be an instance of any one of the following classes:
+                    
+            * `AWS::SDK::Core::StaticCredentialsProvider` - Used for fetching static, non-refreshing
+              credentials.
+
+            * `AWS::SDK::Core::ProcessCredentialsProvider` - Used for loading credentials from a
+              process that outputs JSON to stdout.
+
+            * `AWS::SDK::STS::AssumeRoleCredentialsProvider` - Used when you need to assume a role.
+                    
+            * `AWS::SDK::STS::AssumeRoleWebIdentityCredentialsProvider` - Used when you need to
+              assume a role after providing credentials via the web using a token.
+                    
+            * `AWS::SDK::SSO::RoleCredentialsProvider` - Used for loading credentials from AWS SSO
+              using an access token generated from `aws login`.
+                    
+            * `AWS::SDK::Core::InstanceCredentialsProvider` - Used for loading credentials from the instance
+              metadata service (IMDS) on an EC2 instance.
+                    
+            * `AWS::SDK::Core::ContainerCredentialsProvider - Used for loading credentials from instances
+              running in ECS.
+                    
+            When `:credentials_provider` is not configured directly, the following
+            locations will be searched for credentials:
+                    
+            * ENV['AWS_ACCESS_KEY_ID'], ENV['AWS_SECRET_ACCESS_KEY'], and other
+              ENV variables that influence credentials.
+            * `~/.aws/credentials` and `~/.aws/config`
+            * EC2/ECS instance profiles.
+              
+            @see AWS::SDK::Core::CredentialsProviderChain
+            """;
+    public static final ClientConfig CREDENTIALS_PROVIDER = ClientConfig.builder()
+            .name("credentials_provider")
+            .documentationRbsAndValidationType(Hearth.IDENTITY_PROVIDER.toString())
+            .documentation(CREDENTIALS_PROVIDER_DOCUMENTATION)
+            .defaults(ConfigProviderChain.builder()
+                    // .dynamicProvider("cfg[:credentials_provider_chain]")
+                    .staticProvider("AWS::SDK::Core::CredentialsProviderChain")
+                    .build())
+            .build();
+
+//    private static final String CREDENTIALS_PROVIDER_CHAIN_DOCUMENTATION = """
```

Comment:
What are these for?

Reply:
I tried making the chain configuration but it didn't work. It has to be a literal I think and we don't have support for that. I also don't know if we want to make the chain configurable or not.